activestorage/lib/active_storage/service/azure_storage_service.rb

```diff
@@ -34,16 +34,26 @@ def download(key, &block)
         end
       else
         instrument :download, key: key do
-          _, io = blobs.get_blob(container, key)
-          io.force_encoding(Encoding::BINARY)
+          begin
+            _, io = blobs.get_blob(container, key)
+            io.force_encoding(Encoding::BINARY)
+          rescue Azure::Core::Http::HTTPError => e
+            raise e unless e.type == "BlobNotFound"
```

Comment:
Let’s use bare `raise` instead of `raise e` to avoid munging the backtrace:

```ruby
if e.type == "BlobNotFound"
  raise ActiveStorage::FileNotFoundError
else
  raise
end
```